tests/test_geoprocessing.py

```diff
@@ -4186,3 +4186,40 @@ def test_convolve_2d_nodata(self):
         expected_output[0, 0] = 1
         expected_output[-1, -1] = 1
         numpy.testing.assert_allclose(target_array, expected_output)
+
+    def test_convolve_2d_gaussian(self):
+        """PGP.geoprocessing: test convolve 2d with large gaussian kernel."""
+        # choosing twice the max memory block
+        n_pixels = 256*2
+        # this is a fun seed random seed
+        numpy.random.seed(n_pixels)
+        signal_array = numpy.random.random((n_pixels, n_pixels))

```

Comment:
Nice. It does appear that Numpy frowns upon reseeding the BitGenerator (https://numpy.org/doc/stable/reference/random/generated/numpy.random.seed.html) but, it's probably fine. Also, I'm not sure what numpy version we're sticking to at the moment. It looks like they've recently reworked their random API.

Reply:
I did not know about this! I guess I get why they want to do that -- no global state? I changed it in ad2bf39ea328fe9d71249e1f86f8aefd4ff7a0ed